pkg/cfn/builder/nodegroup.go

```diff
@@ -24,15 +23,25 @@ type NodeGroupResourceSet struct {
 	userData         *gfn.Value
 }
 
-// NewNodeGroupResourceSet returns a resource set for the new node group
-func NewNodeGroupResourceSet(spec *api.ClusterConfig, clusterStackName string, id int) *NodeGroupResourceSet {
+// NewEmbeddedNodeGroupResourceSet returns a resource set for a node group embedded in a cluster config
+func NewEmbeddedNodeGroupResourceSet(spec *api.ClusterConfig, clusterStackName string, ng *api.NodeGroup) *NodeGroupResourceSet {
```

Comment:
I think we should have just one function - `NewNodeGroupResourceSet`.

Reply:
Good catch 👍 Confirmed that you've finished it.